src/error/mod.rs

```diff
@@ -0,0 +1,58 @@
+use std::{error::Error, fmt::Display};
+
+#[derive(Debug, PartialEq)]
+pub enum GetImportErrorKind {
+    ParseTsFileError,
+    FileDoesNotExist,
+    ReadImportError,
+    ReadTsFileError,
+}
+
+#[derive(Debug)]
+pub struct GetImportError {
+    pub kind: GetImportErrorKind,
+    pub parser_errors: Option<Vec<String>>,
+    pub io_errors: Option<Vec<std::io::Error>>,
+    pub file_path: Option<String>,
+}
```

Comment:
Here you're basically reimplementing union member discrimination, but without the typechecking. Rather than have a bunch of optional fields which are populated depending on the value of `kind`, you should make these enum members with fields like it was before

If it gets complicated, I'd suggest declaring different structs for the expected fields of each error kind